experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,296 @@
+flag set, all other flags would be unset. The last frame would have the end flag
+set, all other flags would be unset. Any frames in between would have no flags
+set.
+
+For small messages sent as a single frame, the start flag and the end flag would
+be set, all other flags would be unset.
+
+### Layers Diagram
+
+The definition section defines three layers:
+
+- Method Layers: Defines the available methods, parameter and return value
+  encoding
+- Message Layer: Defines Request Messages and Response Messages
+- Frame Layer: Defines how Messages are split into frames
+
+The diagram below shows a practial example of how a method invocation would look
+like after being encoded as two frames.
+
+#### Frame 0
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         frame padding                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|          frame length         |          frame flags          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                 frame body [ message length ]                 |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|              frame body [ message invocation_id ]             |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                frame body [ message method_id ]               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                 frame body [ message padding ]                |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                                                               +
+|                                                               |
++   frame body [ message body [ method encoded_parameters ] ]   +
+|                                                               |
++                                                               +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ....
+```
+
+#### Frame 1
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         frame padding                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|          frame length         |          frame flags          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++   frame body [ message body [ method encoded_parameters ] ]   +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
For future reference if we need to edit this: It was generated with [Protocol](https://www.luismg.com/protocol/), using the following schema:

`"frame padding:32,frame length:16,frame flags:16,frame body [ message body [ method encoded_parameters ] ]:64"`

thx @tiziano88 for recommending the tool